pygfx/utils/transform.py

```diff
@@ -571,36 +583,81 @@ def __init__(
         self._scale_view = self._scale.view()
         self._scale_view.flags.writeable = False
 
+        # The ._matrix is only used when state_basis is "matrix"
+        if state_basis == "matrix" and matrix is not None:
+            self._matrix = np.asarray(matrix, dtype=float)
+        else:
+            self._matrix = np.identity(4, dtype=float)
+
+        self._matrix_view = self._matrix.view()
+        self._matrix_view.flags.writeable = False
+
     def flag_update(self):
         self.last_modified = perf_counter_ns()
         super().flag_update()
 
+    @property
+    def state_basis(self) -> str:
+        """The basis of the transform, either "components" (default) or "matrix"."""
+        return self._state_basis
+
+    @state_basis.setter
+    def state_basis(self, value):
+        if value not in ("components", "matrix"):
+            raise ValueError("state_basis must be either 'components' or 'matrix'")
+        if value == "matrix":
+            self._matrix[:] = self.matrix
+        elif value == "components":
+            self._position[:] = self.position
+            self._rotation[:] = self.rotation
+            self._scale[:] = self.scale
+        self._state_basis = value
+        self.flag_update()
+
     @property
     def position(self) -> np.ndarray:
-        return self._position_view
+        """The origin of source."""
+        if self.state_basis == "components":
+            return self._position_view
+        return super().position
 
     @position.setter
     def position(self, value):
-        self._position[:] = value
-        self.flag_update()
+        if self.state_basis == "components":
+            self._position[:] = value
+            self.flag_update()
+            return
+        super().position = value
```

Comment:
I'm trying to run the example from #922, but I get `'super' object has no attribute 'position'` for this line.

Reply:
Wow apparently this is a bug in python that's been open for 10+ years: https://bugs.python.org/issue14965

Guess I have to refactor this

Reply:
See #923